dynamiq/callbacks/streaming.py

```diff
@@ -218,3 +266,521 @@ async def __aiter__(self) -> AsyncIterator[StreamingEventMessage]:
+                self._last_emit_index = len(self._buffer)
+
+    def _extract_text_delta(self, chunk: dict[str, Any]) -> str:
+        """Extract textual content from streaming chunk received from the LLM.
+
+        Returns:
+            str: The extracted content.
+        """
+        extracted_content = ""
+
+        choices = chunk.get("choices") or []
+        if choices:
+            delta = choices[0].get("delta", {})
+            content = delta.get("content")
+            if isinstance(content, str):
+                extracted_content = content
+
+        if not extracted_content and isinstance(chunk.get("content"), str):
+            extracted_content = chunk.get("content")
+
+        return extracted_content
+
+    def _extract_function_calling_text(self, chunk: dict[str, Any]) -> tuple[str, str | None]:
+        """
+        Extract incremental JSON values (arguments) and function name f
+        rom the LLM streaming chunks in FUNCTION_CALLING inference mode.
+
+        Returns:
+            tuple[str, str | None]: (arguments_text, function_name)
+        """
+        arguments_text = ""
+        function_name = None
+
+        choices = chunk.get("choices") or []
+        if choices:
+            delta = choices[0].get("delta", {})
+            tool_calls = delta.get("tool_calls")
+
+            if tool_calls and len(tool_calls) > 0:
+                tool_call = tool_calls[0]
+                if tool_call.get("type") == "function" and "function" in tool_call:
+                    function_data = tool_call["function"]
+
+                    if function_data.get("name"):
+                        function_name = function_data["name"]
+
+                    if function_data.get("arguments"):
+                        arguments_text = function_data["arguments"]
+
+        return arguments_text, function_name
+
+    def _emit(self, content: str, step: str) -> None:
+        """Emit the parsed content using the agent's stream_content method.
+
+        Args:
+            content (str): The content to stream.
+            step (str): The step to stream the content to.
+        """
+        if not content:
+            return
+
+        # Skip streaming if in FINAL mode and not in answer step
+        if self.agent.streaming.mode == StreamingMode.FINAL and step != StreamingState.ANSWER:
+            return
+
+        # Format content based on the step type
+        if step == "reasoning":
+            content_to_stream = {
+                "thought": content,
+                "loop_num": self.loop_num,
+            }
+        elif step == "answer":
+            content_to_stream = content
+
+        self.agent.stream_content(
+            content=content_to_stream,
+            source=self.agent.name,
+            step=step,
+            config=self.config,
+            **(self.kwargs | {"loop_num": self.loop_num}),
+        )
+
+    def _process_default_mode(self, final_answer_only: bool) -> None:
+        tags = {
```

Comment:
I don't understand reason of this dict. You could just use `find(DefaultModeTag.THOUGHT)` and  also update this `for tag_text, name in ((tags["action"], "action"), (tags["answer"], "answer")):` to proper usage just DefaultModeTag enum